MapboxCoreNavigation/RouteController.swift

```diff
@@ -585,14 +585,10 @@ extension RouteController: CLLocationManagerDelegate {
     }
     
     func updateRouteLegProgress(for location: CLLocation) {
-        guard routeProgress.currentLegProgress.upComingStep?.maneuverType == .arrive else { return }
-        
-        let polyline = Polyline(routeProgress.currentLegProgress.currentStep.coordinates!)
-        let userSnapToStepDistanceFromManeuver = polyline.distance(from: location.coordinate)
-        let secondsToEndOfStep = userSnapToStepDistanceFromManeuver / location.speed
         let currentDestination = routeProgress.currentLeg.destination
+        let legDurationRemaining = routeProgress.currentLegProgress.durationRemaining
```

Comment:
Slight optimization, we do not need to re-compute the duration remaining for the leg. This is also a little more clear because we should not be comparing step duration when we're interested more in the duration of the leg for knowing when the user arrives.